### ISSUE TITLE
CalendarDatePicker popout displays incorrect date highlight if SelectedDate set through another means.

### ISSUE THREAD

**Issue by StagPoint** [AvaloniaUI/Avalonia#13000]
**Describe the bug**
Once you've used the popout to select a date, it will continue to highlight that date even if you've changed SelectedDate through some other means (such as code-behind event handlers).

**To Reproduce**

1. Open the popout and select a date
2. Change the SelectedDate property through code (in my case, from a button click event handler)
3. Open the popout again
4. You will see that the popout highlights the last date selected through the popout, not the SelectedDate property

**Expected behavior**
The popout should always highlight whichever date is assigned to the SelectedDate property when it is opened.

**Screenshot**
![Avalonia CalendarDatePicker Incorrect Date Highlight](https://github.com/AvaloniaUI/Avalonia/assets/10640987/0c8bf859-d94d-458e-bd21-dbfee65f35ee)

**Comment by Alshain01** [comment]
I have confirmed the bug. 
It only occurs after selecting a date from the calendar.  If you initialize ``CalendarDatePicker.SelectedDate`` in code and only use code (the button) to increment the date, the calendar selection will move appropriately.  Furthermore, this doesn't appear to be an issue in ``Calendar`` itself.  I've created a calendar outside of the CalendarDatePicker and no amount of clicking a date and then using a button to advance it causes the same issue.

This occurs with both the Simple and Fluent themes.
